core/file_walker.go

```diff
@@ -1,50 +1,70 @@
 package core
 
 import (
-	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 )
 
 type File struct {
 	Name  string
+	Dir   *File
```

Comment:
There are 2 ways how to do this. Either we will include pointer on `Parent` here as you did, or we will use the `State.ancestors` to compute path to a marked file (something like `State.Path(`) is doing) The benefit of your solution is that you can walk the folder tree in both directions.

Benefit of the alternative is that you still keep the `File` struct very simple and the tests including `File` are a bit simpler. Specially because if we wanted the test cases to reflect reality we couldn't put `nil` as a parent pointer we would have to put there the correct pattern.

What do you think about this @leonklingele?